### DISCUSSION TITLE
Azure pipeline - IOS - no such module 'Capacitor'

### DISCUSSION THREAD

**Discussion by Boimb** [ionic-team/capacitor#6717] [Q&A] [answered]
Hi Everyone,
I'm new to Ionic and Capacitor.
Posting my question here because discussions are closed in the framework repo and I won't say it's a bug, more a lack of understanding on my side.

I just followed the "Build your first app" tutorial on the ionic framework page. Everything goes as expected with the SPA and Android apps that I can launch on my local.

But to try a build on IOS, I rely on Azure pipelines and here comes the issue.
Basically, the `Xcode@5` build task fails with this message:

```bash
❌  /Users/runner/work/1/s/ios/App/App/AppDelegate.swift:2:8: no such module 'Capacitor'
import Capacitor
```

Here's my pipeline conf:
```azure-pipelines.yaml
trigger:
  - chore/pipeline

pool:
  vmImage: "macos-latest"

steps:
  - task: Npm@1
    inputs:
      command: "ci"

  - task: Npm@1
    inputs:
      command: custom
      customCommand: run build

  - bash: |
      sudo npm i -g @ionic/cli

  - bash: |
      npx ionic cap sync

  - task: InstallAppleCertificate@2
    inputs:
      certSecureFile: ios_development.p12
      certPwd: $(certPwd)
      keychain: "temp"

  - task: InstallAppleProvisioningProfile@1
    inputs:
      provisioningProfileLocation: "secureFiles"
      provProfileSecureFile: iosapp.mobileprovision

  - task: CocoaPods@0
    inputs:
      workingDirectory: "$(Build.SourcesDirectory)/ios/App"
      forceRepoUpdate: false

  - task: Xcode@5
    inputs:
      actions: "build"
      scheme: ""
      sdk: "iphoneos"
      configuration: "Debug"
      xcWorkspacePath: "**/*.xcodeproj/project.xcworkspace"
      xcodeVersion: "default"
      packageApp: true
      signingOption: "nosign"
```

For now, I don't sign the app cause had issues with that too, but let's try to solve probs one at a time ^^ 
Thx for your support.

**Comment by mlynch** [comment]
Hi @Boimb, hard to say, looks like the pod dependencies aren't being loaded correctly. Unfortunately, there's not much we can do to help given we don't control that service. I'd like to throw in a mention for our [Appflow](https://ionic.io/appflow) build service. It [integrates with Azure DevOps](https://ionic.io/docs/appflow/quickstart/azure-devops) and was built specifically with Capacitor apps in mind. It's the best cloud build service for Capacitor apps. Happy to help with anything if you end up checking it out.

**Comment by Boimb** [comment, marked as answer]
Hi @mlynch, thanks for the answer.
Yes, your builtin solution/saas build service seems super nice. Unfortunately, I have to work with resources I have for the moment.

For others that might hit this discussion, I got the fix (don't know how I missed it on [SO](https://stackoverflow.com/questions/72793699/xcode-build-with-azure-pipelines-for-ionic-capacitor-app-no-such-module-capaci) :/ )
In the build task, you have to specify the `App.xcworkspace` path, not the `project.xcworkspace` (as mentioned in the MS doc)

```yaml
      xcWorkspacePath: '$(Build.SourcesDirectory)/ios/App/App.xcworkspace'
```
